### DISCUSSION TITLE
How to hide Cash App Pay and Klarna from Flutter Stripe payment sheet

### DISCUSSION THREAD

**Discussion by alishaircodercrew** [flutter-stripe/flutter_stripe#1665] [Q&A]
![Uploading stripe.png…]()
Map<String, dynamic> _paymentIntent = {};
This line declares a map _paymentIntent that will hold the payment intent information fetched from the server. It's initially empty.

        Future<void> makePayment({
          required BuildContext context,
          required Dio paymentDio,
          int? total,
        }) async {
          try {
            _paymentIntent = await createPaymentIntent('$total', 'usd', paymentDio);

            //STEP 2: Initialize Payment Sheet
            await Stripe.instance
                .initPaymentSheet(
                paymentSheetParameters: SetupPaymentSheetParameters(
                    paymentIntentClientSecret: _paymentIntent['client_secret'],
                    style: ThemeMode.light,
                    merchantDisplayName: 'Ikay',
                ))
                .then((value) {});

            //STEP 3: Display Payment sheet
            await displayPaymentSheet(context: context);
          } catch (err) {
            print("error is $err");
            throw Exception(err);
          }
        }
This function makePayment initiates the payment process. It first creates a payment intent using the createPaymentIntent function, then initializes the Payment Sheet using the obtained client secret. Finally, it displays the Payment Sheet.

        Future<void> displayPaymentSheet({
          required BuildContext context,
        }) async {
          try {
            await Stripe.instance.presentPaymentSheet().then((value) {
              showDialog(
                  context: context,
                  builder: (_) => const AlertDialog(
                    content: Column(
                      mainAxisSize: MainAxisSize.min,
                      children: [
                        Icon(
                          Icons.check_circle,
                          color: Colors.green,
                          size: 100.0,
                        ),
                        SizedBox(height: 10.0),
                        Text("Payment Successful!"),
                      ],
                    ),
                  ));
              // getOrderDetails(context, dio);

              _paymentIntent = {};
            }).onError((error, stackTrace) {
              throw Exception(error);
            });
          } on StripeException catch (e) {
            print('Error is:---> $e');
            const AlertDialog(
              content: Column(
                mainAxisSize: MainAxisSize.min,
                children: [
                  Row(
                    children: [
                      Icon(
                        Icons.cancel,
                        color: Colors.red,
                      ),
                      Text("Payment Failed"),
                    ],
                  ),
                ],
              ),
            );
          } catch (e) {
            print('Exception is: $e');
          }
        }
This function displayPaymentSheet actually presents the Payment Sheet to the user. It shows a dialog indicating whether the payment was successful or failed based on the response from the Payment Sheet.

**Comment by Bill550** [comment]
@alishaircodercrew did you find any solution ?

**Comment by Nikunj-Flutter** [comment]
You can disable the cash app from the stripe dashboard [https://dashboard.stripe.com/](url)
